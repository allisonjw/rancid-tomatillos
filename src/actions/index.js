export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies
});

export const getRatings = ratings => ({
  type: 'GET_RATINGS',
  ratings
});

export const addRating = rating => ({
  type: 'ADD_RATING',
  rating
});

export const getUser = user => ({
  type: 'ADD_USER',
  user
});

export const deleteRating = rating => ({
  type: 'REMOVE_RATING',
  rating
});

export const logOut = () => ({
  type: 'LOG_OUT',
})

export const setError = (message) => ({
  type: 'HAS_ERROR',
  message
})