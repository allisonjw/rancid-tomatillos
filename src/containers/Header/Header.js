import React from 'react';
import { connect } from 'react-redux';
import { logOut } from '../../actions/index'
import { NavLink } from 'react-router-dom';
import logo from '../../util/images/tomatoe-logo copy.png';
import textLogo from '../../util/images/rancid-tom-white.png';

export const Header = ({user, logUserOut}) => {
  const username = user.name;
  return(
    <div className="header-div">
      <img className="header-text-logo" src={textLogo} />
      <img className="logo_image" alt="tomato logo" src={logo} />
      {console.log("USER", user.name)}
     {!username ?
        <NavLink to='/login' className="btn login-btn">Login</NavLink> :
        <>
        <NavLink to='/' className="btn logout" onClick={logUserOut}>Logout</NavLink>
        <h2 className="header-username">{`Hello, ${username}!`}</h2>
        </> }
    </div>
  )

}

export const mapStateToProps = state => ({
    user: state.user
  })

export const mapDispatchToProps = dispatch => ({
    logUserOut: () => dispatch(logOut())
})

export default connect(mapStateToProps, mapDispatchToProps)(Header)
