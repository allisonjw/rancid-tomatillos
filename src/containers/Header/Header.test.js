import React from 'react';
import { shallow } from 'enzyme';
import { logOut } from '../../actions/index'
import { Header, mapDispatchToProps } from './Header';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header
      user={{id: 7, name: 'Diana', email: "diana@turing.io"}}
      logUserOut={jest.fn()}
    />);
  })
  it('should match the snapshot with correct data', () => {

    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot with correct data when user is logged out', () => {
    const wrapper = shallow(<Header
      user={{}}
      logUserOut={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call logUserOut when button is clicked', () => {
    const logOutMock = jest.fn()
    const wrapper = shallow(<Header
      user={{id: 7, name: 'Diana', email: "diana@turing.io"}}
      logUserOut={logOutMock}
    />);

    wrapper.find(".logout").simulate('click')

    expect(logOutMock).toHaveBeenCalled()

  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct action when logUserOut is called', () => {
      const mockDispatch = jest.fn();
      const dispatchedAction = logOut();

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.logUserOut()

      expect(mockDispatch).toHaveBeenCalledWith(dispatchedAction)
    })
  })
});
