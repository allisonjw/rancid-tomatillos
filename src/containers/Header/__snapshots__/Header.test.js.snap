// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with correct data 1`] = `
<div
  className="header-div"
>
  <h1
    className="header-app-name"
  >
    Rancid Tomatillos
  </h1>
  <NavLink
    className="btn logout"
    onClick={[MockFunction]}
    to="/"
  >
    Logout
  </NavLink>
  <h2
    className="header-username"
  >
    Hello, Diana!
  </h2>
</div>
`;

exports[`Header should match the snapshot with correct data when user is logged out 1`] = `
<div
  className="header-div"
>
  <h1
    className="header-app-name"
  >
    Rancid Tomatillos
  </h1>
  <NavLink
    className="login-btn"
    to="/login"
  >
    Login
  </NavLink>
</div>
`;
